src/algokit/cli/generate.py

```diff
@@ -0,0 +1,121 @@
+import json
+import logging
+import pathlib
+import platform
+import re
+
+import algokit_client_generator
+import click
+
+from algokit.core import proc
+
+logger = logging.getLogger(__name__)
+
+
+def snake_case(s: str) -> str:
+    s = s.replace("-", " ")
+    s = re.sub(r"([A-Z]+)([A-Z][a-z])", r"\1_\2", s)
+    s = re.sub(r"([a-z\d])([A-Z])", r"\1_\2", s)
+    return re.sub(r"[-\s]", "_", s).lower()
+
+
+def format_client_name(output: pathlib.Path, application_file: pathlib.Path) -> pathlib.Path:
+    client_name = str(output).replace("%parent_dir%", snake_case(application_file.parent.name))
+
+    if "%name%" in str(output):
+        application_json = json.loads(application_file.read_text())
+        client_name = str(output).replace("%name%", snake_case(application_json["contract"]["name"]))
+
+    return pathlib.Path(client_name)
+
+
+def generate_client_by_language(app_spec: pathlib.Path, output: pathlib.Path, language: str) -> None:
+    if language.lower() == "typescript":
+        is_windows = platform.system() == "Windows"
+        npx = "npx" if not is_windows else "npx.cmd"
+        cmd = [
+            npx,
+            "--yes",
+            "@algorandfoundation/algokit-client-generator@v2.0.0-beta.1",
+            "generate",
+            "-a",
+            str(app_spec),
+            "-o",
+            str(output),
+        ]
+        try:
+            proc.run(
+                cmd,
+                bad_return_code_error_message=f"Failed to run {' '.join(cmd)} for {app_spec}.",
+            )
+        except OSError as e:
+            raise click.ClickException("Typescript generator requires Node.js and npx to be installed.") from e
+        logger.info(
+            f"Generating TypeScript client code for application specified in {app_spec} and writing to {output}"
+        )
+    elif language.lower() == "python":
```

Comment:
Putting the shorter branch of a conditional first can aid in readability since it uses less "mental stack space" that need to be "unwound" once you reach the end of a branch, so I would suggest flipping this around